src/libraries/System.Text.Json/src/System/Text/Json/Serialization/MemberAccessor.cs

```diff
@@ -28,5 +28,20 @@ internal abstract class MemberAccessor
         public abstract Func<object, TProperty> CreateFieldGetter<TProperty>(FieldInfo fieldInfo);
 
         public abstract Action<object, TProperty> CreateFieldSetter<TProperty>(FieldInfo fieldInfo);
+
+#if DEBUG
+        public static bool HasConstructor(Type type, ConstructorInfo constructor)
+        {
+            foreach (ConstructorInfo candidate in type.GetConstructors(BindingFlags.Public | BindingFlags.Instance))
+            {
+                if (ReferenceEquals(candidate, constructor))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
```

Comment:
What you have is fine, but could this just be:
```C#
public static bool HasConstructor(Type type, ConstructorInfo constructor) =>
    Array.IndexOf(type.GetConstructors(BindingFlags.Public | BindingFlags.Instance), constructor) >= 0;
```
?

Reply:
Thanks